modin/experimental/core/execution/native/implementations/hdk_on_native/dataframe/dataframe.py

```diff
@@ -2842,13 +2845,13 @@ def from_arrow(
 
         for col in at.columns:
             if pyarrow.types.is_dictionary(col.type):
-                new_dtypes.append(
-                    LazyProxyCategoricalDtype._build_proxy(
-                        parent=at,
-                        column_name=col._name,
-                        materializer=build_categorical_from_at,
-                    )
+                dt = LazyProxyCategoricalDtype._build_proxy(
+                    parent=at,
+                    column_name=col._name,
+                    materializer=build_categorical_from_at,
                 )
+                set_cat_value_dtype(dt, arrow_type_to_pandas(col.type.value_type))
```

Comment:
why dtype can't be a parameter here?
```suggestion
                dt = LazyProxyCategoricalDtype._build_proxy(
                    parent=at,
                    column_name=col._name,
                    materializer=build_categorical_from_at,
                    dtype=arrow_type_to_pandas(col.type.value_type),
                )
```

Reply:
Just to avoid modifications of the LazyProxyCategoricalDtype class, which is used by other backends. I'm not sure if this parameter could be useful for other backends.

Reply:
Will it require a lot of changes in `LazyProxyCategoricalDtype`? I believe that the changes could fit into something like:
```diff
diff --git a/modin/core/dataframe/pandas/metadata/dtypes.py b/modin/core/dataframe/pandas/metadata/dtypes.py
index c249a3fb..a6a3b0e9 100644
--- a/modin/core/dataframe/pandas/metadata/dtypes.py
+++ b/modin/core/dataframe/pandas/metadata/dtypes.py
@@ -199,10 +199,10 @@ class LazyProxyCategoricalDtype(pandas.CategoricalDtype):
         elif parent is self._parent and column_name == self._column_name:
             return self
         else:
-            return self._build_proxy(parent, column_name, self._materializer)
+            return self._build_proxy(parent, column_name, self._materializer, self._dtype)
 
     @classmethod
-    def _build_proxy(cls, parent, column_name, materializer):
+    def _build_proxy(cls, parent, column_name, materializer, dtype=None):
         """
         Construct a lazy proxy.
 
@@ -223,8 +223,14 @@ class LazyProxyCategoricalDtype(pandas.CategoricalDtype):
         result._parent = parent
         result._column_name = column_name
         result._materializer = materializer
+        result._dtype = dtype
         return result
 
+    def _get_dtype(self):
+        if self._dtype is None:
+            self._dtype = self.categories.dtype
+        return self._dtype
+
     def __reduce__(self):
         """
         Serialize an object of this class.
@@ -265,6 +271,7 @@ class LazyProxyCategoricalDtype(pandas.CategoricalDtype):
         self._categories_val = categories
         self._parent = None  # The parent is not required any more
         self._materializer = None
+        self._dtype = None
 
     @property
     def _is_materialized(self) -> bool:
```
In this implementation, it's suggested, that consumers will get `dtype` using `._get_dtype()` method. This way, we encapsulate the dtype handling logic inside the class and ensure that it's properly copied to new instances of the proxy and also that the cache is being reset when new categories are being set.